### PR TITLE
Create stories for Source v4

### PR DESCRIPTION
**Pull request by jamie-lynch**
## What is the purpose of this change?

This change adds stories for the v4 components so that the documentation is clear for users of the new `source-*` packages.

## What does this change?

- Update the title of kitchen stories to be `Source v4/source-react-components-development-kitchen/*`  
- Create story files for v4 that re-export everything from their v3 equivalents but with the correct v4 title
- Add a script to compare and sync the v3 and v4 stories
- Add a CI check to run the story comparison

### CODE REVIEW

File: scripts/story-sync/compare.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const v3StoriesNotInV4 = v3Stories.filter(
+	(story) => !v4Stories.includes(story),
+);
+
+const v4StoriesNotInV3 = v4Stories.filter(
+	(story) => !v3Stories.includes(story),
+);
+
+if (v3StoriesNotInV4.length || v4StoriesNotInV3.length) {
+	console.error('v4 and v4 stories are out of sync');
```
Comment (SiAdcock) [inline scripts/story-sync/compare.ts:17]:
```suggestion
	console.error('v3 and v4 stories are out of sync');
```

File: scripts/story-sync/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "compare": "ts-node ./compare.ts",
+    "sync": "ts-node ./sync.ts"
+  },
+  "dependencies": {
+    "react-scanner": "^1.0.0",
+    "rimraf": "^3.0.2"
+  },
+  "devDependencies": {
+    "@types/rimraf": "^3.0.2"
+  }
```
Comment (SiAdcock) [inline scripts/story-sync/package.json:17]: It doesn't look like we're using these deps / devDeps

File: packages/@guardian/source-react-components-development-kitchen/src/components/divider/Divider.stories.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import type { DividerProps } from './index';
 import { Divider } from './index';
 
 export default {
-	title: 'Kitchen/source-react-components-development-kitchen/Divider',
+	title: 'Source v4/source-react-components-development-kitchen/Divider',
```
Comment (SiAdcock) [inline packages/@guardian/source-react-components-development-kitchen/src/components/divider/Divider.stories.tsx:10]: Maybe a pedantic point, but I wouldn't say the dev kitchen falls under "Source v4". I do see where you're coming from though 🤔  Will this make navigation simpler or content more discoverable?

Comment (SiAdcock) [reply packages/@guardian/source-react-components-development-kitchen/src/components/divider/Divider.stories.tsx:10]: A tangential point. When I created the Foundations docs, I (probably mistakenly) did not organise them under a package. I think we can address this, either here or separately, by updating the title of Foundations docs from:

```tsx
<Meta
	title="Foundations/Colour"
	parameters={{ previewTabs: { canvas: { hidden: true } } }}
/>
```

```tsx
<Meta
	title="Source v4/source-foundations/Colour"
	parameters={{ previewTabs: { canvas: { hidden: true } } }}
/>
```
e.g. [colour.stories.mdx](https://raw.githubusercontent.com/guardian/source/main/packages/%40guardian/source-foundations/src/colour.stories.mdx)

File: scripts/story-sync/package.json (hunk 2 above)
Comment (jamie-lynch) [reply scripts/story-sync/package.json:17]: Good point! That's a classic copy-and-paste error. 🤦

File: packages/@guardian/source-react-components-development-kitchen/src/components/divider/Divider.stories.tsx (hunk 3 above)
Comment (jamie-lynch) [reply packages/@guardian/source-react-components-development-kitchen/src/components/divider/Divider.stories.tsx:10]: Agreed. I've dropped the commit that changes the kitchen stories. Let's raise a separate PR that updates foundations.

**Review by SiAdcock — APPROVED** [review]
Amazing work @jamie-lynch 💎 ❤️ 👍

**Pull request merged**